platform/mellanox/mlnx-platform-api/sonic_platform/fan.py

```diff
@@ -352,3 +385,36 @@ def get_speed_tolerance(self):
         """
         # The tolerance value is fixed as 20% for all the Mellanox platform
         return 20
+
+    @classmethod
+    def set_cooling_level(cls, level):
+        """
+        Change cooling level. The input level should be an integer value [1, 10].
+        1 means 10%, 2 means 20%, 10 means 100%.
+        """
+        if not isinstance(level, int):
+            raise RuntimeError("Failed to set cooling level, input parameter must be integer")
+
+        if level < 1 or level > 10:
+            raise RuntimeError("Failed to set cooling level, level value must be in range [1, 10], got {}".format(level))
+
+        try:
+            # reset FAN driver and change cooling state
+            with open(COOLING_STATE_PATH, 'w') as cooling_state:
+                cooling_state.write(str(level + 10))
```

Comment:
can you please clarify? how do you reset with just setting the cooling level?

Reply:
Added comment.